test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -2260,9 +2280,13 @@ TEST_P(SslSocketTest, FailedClientCertificateSpkiVerificationWrongClientCertific
   client_cert->mutable_private_key()->set_filename(TestEnvironment::substitute(
       "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/no_san_key.pem"));
 
-  TestUtilOptionsV2 test_options(listener, client, false, GetParam());
-  testUtilV2(test_options.setExpectedServerStats("ssl.fail_verify_cert_hash")
-                 .setExpectedTransportFailureReasonContains("SSLV3_ALERT_CERTIFICATE_UNKNOWN"));
+  TestUtilOptionsV2 test_options(listener, client, false, version_);
+  testUtilV2(
+      test_options.setExpectedServerStats("ssl.fail_verify_cert_hash")
+          .setExpectedTransportFailureReasonContains(
+              Runtime::runtimeFeatureEnabled("envoy.reloadable_features.tls_async_cert_validation")
```

Comment:
Why is the failure reason different in async vs not-async cases?

Reply:
This test doesn't test async validation, but switching to the new interface allows us to populate the boring ssl out_alert in the callback, so we can provide more specific error to boring stack.